### DISCUSSION TITLE
Is it possible to do an on-the-fly mirror?

### DISCUSSION THREAD

**Discussion by flixman** [gitpython-developers/GitPython#1586] [Q&A]
I have a repository that I want to mirror. Normally I'd fork it and then do periodic pulls from upstream, but these two repos happen to be in AzureDevOps, and this is not possible. I was, then, thinking in having a script that queries the upstream repo, gets the tree, blob, commit and tags objects, and pushes them into the mirror. Do you guys think this approach would work? Ideally I'd like to do this on a service that does not have even disk (like listening/querying for changes and then retrieving the changed objects).

Is this possible at all?

**Comment by Byron** [comment]
If that would be possible with the `git` CLI then it would be possible with GitPython. Working purely in-memory is nothing I think any implementation can truly due, except for maybe `isomorphic-git` as it is meant to run in the browser. Maybe `pygit2` might be usable here though, as `libgit2` is the most mature `git` implementation I know (and if it would have been as mature 10 years ago, GitPython probably wouldn't even exist today).

**Reply by flixman** [reply]
ok! I will check that route! Thank you! :-)
